generic3g/specs/VariableSpec.F90

```diff
@@ -224,4 +205,97 @@ function get_regrid_method_from_field_dict_(standard_name, rc) result(regrid_met
       _RETURN(_SUCCESS)
    end function get_regrid_method_from_field_dict_
 
+   function make_VariableSpecFromAspects(state_intent, short_name, unusable, &
+         & standard_name, itemtype, service_items, default_value, bracket_size, &
+         & dependencies, geom_aspect, units_aspect, attributes_aspect, &
+         & ungridded_aspect, vertical_aspect, frequency_aspect, typekind_aspect, rc) &
+         & result(var_spec)
+      type(VariableSpec) :: var_spec
+      type(ESMF_StateIntent_Flag), intent(in) :: state_intent
+      character(*), intent(in) :: short_name
+      class(KeywordEnforcer), optional, intent(in) :: unusable
+      character(*), optional, intent(in) :: standard_name
+      type(ESMF_StateItem_Flag), optional, intent(in) :: itemtype
+      type(StringVector), optional :: service_items
+      real, optional, intent(in) :: default_value
+      integer, optional, intent(in) :: bracket_size
```

Comment:
you are missing `ClassAspect` which will eliminate several remaining arguments.  E.g., `bracket_size` is part of `BracketClassAspect`  Likewise, I think `service_items` is part of either Wildcard or Service.

Possibly some of this is complicated by the fact that `make_ItemSpec` is still doing some work downstream.   So maybe there is related refactoring to clean this up.   Or it may not make sense.  Give it some thought and then discuss with me.

Reply:
Yes. I was wondering about the other Aspects. They weren't in **makeVariableSpec**, so I ignored them.